packages/ethereum-group-pcd/src/EthereumGroupPCD.ts

```diff
@@ -0,0 +1,334 @@
+   */
+  ethereumGroupProof: string;
+}
+
+export class EthereumGroupPCD
+  implements PCD<EthereumGroupPCDClaim, EthereumGroupPCDProof>
+{
+  type = EthereumGroupPCDTypeName;
+  claim: EthereumGroupPCDClaim;
+  proof: EthereumGroupPCDProof;
+  id: string;
+
+  public constructor(
+    id: string,
+    claim: EthereumGroupPCDClaim,
+    proof: EthereumGroupPCDProof
+  ) {
+    this.id = id;
+    this.claim = claim;
+    this.proof = proof;
+  }
+}
+
+let addrMembershipConfig: ProverConfig = {
+  circuit: __dirname.concat("/../artifacts/addr_membership.circuit"),
+  witnessGenWasm: __dirname.concat("/../artifacts/addr_membership.wasm"),
+};
+let pubkeyMembershipConfig: ProverConfig = {
+  circuit: __dirname.concat("/../artifacts/pubkey_membership.circuit"),
+  witnessGenWasm: __dirname.concat("/../artifacts/pubkey_membership.wasm"),
+};
+let addrProver: MembershipProver;
+let pubkeyProver: MembershipProver;
+let addrVerifier: MembershipVerifier;
+let pubkeyVerifier: MembershipVerifier;
+
+export async function init(args: EthereumGroupPCDInitArgs): Promise<void> {
+  addrMembershipConfig = args.addrMembershipConfig ?? addrMembershipConfig;
+  pubkeyMembershipConfig =
+    args.pubkeyMembershipConfig ?? pubkeyMembershipConfig;
+  addrProver = new MembershipProver(addrMembershipConfig);
+  pubkeyProver = new MembershipProver(pubkeyMembershipConfig);
+  addrVerifier = new MembershipVerifier(addrMembershipConfig);
+  pubkeyVerifier = new MembershipVerifier(pubkeyMembershipConfig);
+  await Promise.all([
+    addrProver.initWasm(),
+    pubkeyProver.initWasm(),
+    addrVerifier.initWasm(),
+    pubkeyVerifier.initWasm(),
+  ]);
+  return SemaphoreSignaturePCDPackage.init!(args);
+}
+
+/**
+ * Utility function to get the raw public key buffer from a hex string.
+ *
+ * The offset that the hex representation of the public key starts at, without the 0x prefix and without the 04 encoding prefix. That's what the spartan-ecdsa circuit expects.
+ *
+ * https://github.com/indutny/elliptic/issues/86
+ *
+ * https://dev.to/q9/finally-understanding-ethereum-accounts-1kpe
+ */
+export function getRawPubKeyBuffer(pubKey: string): Buffer {
+  if (pubKey.length !== 132) {
+    throw new Error(
+      `invalid public key length ${pubKey.length}. Expected 130 (hex string)`
+    );
+  }
+  const hexPubkeyOffset = 2 + 2;
+  return Buffer.from(pubKey.slice(hexPubkeyOffset), "hex");
+}
+
+export async function prove(
+  args: EthereumGroupPCDArgs
+): Promise<EthereumGroupPCD> {
+  if (args.identity.value === undefined) {
+    throw new Error(`missing argument identity`);
+  }
+
+  if (args.signatureOfIdentityCommitment.value === undefined) {
+    throw new Error(`missing argument signatureOfIdentityCommitment`);
+  }
+
+  if (args.merkleProof.value === undefined) {
+    throw new Error(`missing argument merkleProof`);
+  }
+
+  if (
+    [GroupType.ADDRESS, GroupType.PUBLICKEY].includes(args.groupType as any)
```

Comment:
```suggestion
    ![GroupType.ADDRESS, GroupType.PUBLICKEY].includes(args.groupType as any)
```

you should run all your code :P

Reply:
ah.. it was a double error. Missing the `!` and missing the `.value`, which led to tests passing.